### ISSUE TITLE
Warning (unused label) when generating coverage info

### ISSUE THREAD

**Issue by pitrou** [cython/cython#2974]
This is a regression following #2938 in this function:
```cpp
static PyCodeObject *__Pyx_createFrameCodeObject(const char *funcname, const char *srcfile, int firstlineno) {
    PyObject *py_srcfile = 0;
    PyObject *py_funcname = 0;
    PyCodeObject *py_code = 0;
#if PY_MAJOR_VERSION >= 3
    py_code = PyCode_NewEmpty(srcfile, funcname, firstlineno);
    py_code->co_flags |= CO_OPTIMIZED | CO_NEWLOCALS;
#else
    py_funcname = PyString_FromString(funcname);
    py_srcfile = PyString_FromString(srcfile);
    py_code = PyCode_New(
        0,
        0,
        0,
        CO_OPTIMIZED | CO_NEWLOCALS,
        __pyx_empty_bytes,     /*PyObject *code,*/
        __pyx_empty_tuple,     /*PyObject *consts,*/
        __pyx_empty_tuple,     /*PyObject *names,*/
        __pyx_empty_tuple,     /*PyObject *varnames,*/
        __pyx_empty_tuple,     /*PyObject *freevars,*/
        __pyx_empty_tuple,     /*PyObject *cellvars,*/
        py_srcfile,       /*PyObject *filename,*/
        py_funcname,      /*PyObject *name,*/
        firstlineno,
        __pyx_empty_bytes      /*PyObject *lnotab*/
    );
bad:
    Py_XDECREF(py_srcfile);
    Py_XDECREF(py_funcname);
#endif
    return py_code;
}
#endif
```

This produces the following warning (which may then be treated as error):
```
/home/travis/build/apache/arrow/python/build/temp.linux-x86_64-2.7/_orc.cpp: In function 'PyCodeObject* __Pyx_createFrameCodeObject(const char*, const char*, int)':
/home/travis/build/apache/arrow/python/build/temp.linux-x86_64-2.7/_orc.cpp:5770:1: error: label 'bad' defined but not used [-Werror=unused-label]
 bad:
 ^~~
```

**Comment by scoder** [comment]
It's actually even worse. The commit removed the error handling completely. This is fixed in 0.29.9, see d8dbd7609283bcccf52f359da550bc6d5d29bd3e

**Comment by pitrou** [comment]
Wow, thanks for the quick turnaround @scoder :-)
